streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java

```diff
@@ -132,14 +139,8 @@ public void init(ProcessorContext context, StateStore root) {
         segments.openExisting(this.context);
 
         // register and possibly restore the state from the logs
-        context.register(root, new StateRestoreCallback() {
-            @Override
-            public void restore(byte[] key, byte[] value) {
-                put(Bytes.wrap(key), value);
-            }
-        });
+        context.register(root, new RocksDBSegmentsBatchingRestoreCallback());
 
-        flush();
```

Comment:
@guozhangwang I am not sure why flush was needed in the original implementation.

Reply:
Me neither. @guozhangwang can you confirm that it's ok to remove this?

Reply:
Checked on the source code history, I think it is piggy backed from the old manner, in which we do restoration along with the registration of the store (before 0.11.0) whereas now we only register the store but delay restoration in main loop. So I think we can safely remove this.

Reply:
Thanks for confirming @guozhangwang !